### PR TITLE
Add keyproperty to category

### PR DESCRIPTION
**Pull request by acbeni**
closes issue  #1398

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java [hunk 1]
```diff
@@ -63,6 +63,15 @@
 @HasQueryModel(baseInterfaces = {"io.sphere.sdk.queries.QueryModel<io.sphere.sdk.categories.Category>"})
 public interface Category extends Resource<Category>, WithLocalizedSlug, MetaAttributes, Custom {
 
+
+    /**
+     * User-specific unique identifier for the payment.
```
Comment (selo-ecube) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java]: "for the payment" seems to be a copy-and-paste error?

Comment (acbeni) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/Category.java]: yep its a copy past error :/
i ll correct it right away.

File: commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetKey.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @Nullable
+    public String getKey() {
+        return key;
+    }
+
+    public static SetKey of(@Nullable final String key) {
+        return new SetKey(key);
+    }
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/categories/commands/updateactions/SetKey.java]: To improve the consistency here, we should add a `ofUnset()` static method here.

File: commercetools-models/src/main/java/io/sphere/sdk/customers/Customer.java [hunk 3]
```diff
@@ -431,6 +439,17 @@ public String toString() {
...
+
+    /**
+     * Customer’s salutation
+     *
+     * @see io.sphere.sdk.customers.commands.updateactions.SetSalutation
+     *
+     * @return salutation
+     */
+    @Nullable
+    String getSalutation();
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customers/Customer.java]: Nice that you added javadoc here 👍

File: commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java [hunk 4]
```diff
@@ -30,4 +31,26 @@ public void referenceExpansion() throws Exception {
         assertThat(deletedCategory.getParent().getObj()).isNotNull().isEqualTo(parent);
         client().executeBlocking(CategoryDeleteCommand.of(parent));
     }
+
+    @Test
+    public void executionForDeletionByKey() throws Exception {
+        final String key = randomKey();
+        final CategoryDraft draft = CategoryDraftBuilder.of(randomSlug(), randomSlug()).key(key).build();
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java]: How about replacing the first `randomSlug()` call to a new method `SphereTestUtils.randomLocalizedString()`?

File: commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java [hunk 5]
```diff
@@ -30,4 +31,26 @@ public void referenceExpansion() throws Exception {
...
+        final Category category = client().executeBlocking(CategoryCreateCommand.of(draft));
+
+        client().executeBlocking(CategoryDeleteCommand.ofKey(key,category.getVersion()));
+        assertThat(client().executeBlocking(CategoryQuery.of().byId(category.getId())).head()).isEmpty();
+    }
+
+    @Test
+    public void referenceExpansionForDeletionByKey() throws Exception {
+        final String key = randomKey();
+        final Category parent = client().executeBlocking(CategoryCreateCommand.of(CategoryDraftBuilder.of(randomSlug(), randomSlug()).build()));
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java]: How about replacing the first `randomSlug()` call to a new method `SphereTestUtils.randomLocalizedString()`?

File: commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java [hunk 6]
```diff
@@ -30,4 +31,26 @@ public void referenceExpansion() throws Exception {
...
+        final Category category = client().executeBlocking(CategoryCreateCommand.of(CategoryDraftBuilder.of(randomSlug(), randomSlug()).key(key).parent(parent).build()));
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/categories/commands/CategoryDeleteCommandIntegrationTest.java]: How about replacing the first `randomSlug()` call to a new method `SphereTestUtils.randomLocalizedString()`?

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasNoUpdateAction.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.SOURCE)
+@Target({ElementType.METHOD})
+/**
+ * This annotation marks methods that will be ignored while generating update actions with {@link HasUpdateActions}
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasNoUpdateAction.java]: I like your idea to have this annotation. And great that you added javadoc too 👍

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasNoUpdateAction.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package io.sphere.sdk.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.SOURCE)
+@Target({ElementType.METHOD})
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasNoUpdateAction.java]: 👍

**Pull request merged**